src/Components/Address/ShippingAddressForm.tsx

```diff
@@ -72,18 +76,26 @@ interface SettingsShippingAddressFormProps {
   address?: {
     internalID: string
     isDefault: boolean
-    attributes: Address
-  }
+    attributes: ShippingAddress
+  } | null
+  onSuccess?: (address: {
+    internalID?: string
+    attributes?: ShippingAddress
+  }) => void
+  onDelete?: (id: string) => void
 }
 
-export const SettingsShippingAddressForm: FC<SettingsShippingAddressFormProps> = ({
+export const ShippingAddressForm: FC<SettingsShippingAddressFormProps> = ({
   onClose,
   address,
+  onSuccess,
+  onDelete,
 }) => {
   const { submitMutation: submitAddAddress } = useAddAddress()
   const { submitMutation: submitEditAddress } = useEditAddress()
   const { submitMutation: submitSetDefaultAddress } = useSetDefaultAddress()
   const { sendToast } = useToasts()
+  const [showDeleteDialog, setShowDeleteDialog] = useState<boolean>(false)
```

Comment:
we don't need to specify type `<boolean>` here, since we have an initial value and it's inferred